File: solutions/Python/1207.UniqueNumberOfOccurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occurrences = {}
        for number in arr:
          occurrences[number] = occurrences.get(number, 0) + 1
        values = list(occurrences.values())
        return len(set(values)) == len(values)

File: solutions/TypeScript/1207.UniqueNumberOfOccurrences.ts
function uniqueOccurrences(arr: number[]): boolean {
  let occurrences: { [key: number]: number } = {};
  for (let number of arr) {
    occurrences[number] = (occurrences[number] || 0) + 1;
  }
  const values = Object.values(occurrences);
  return new Set(values).size === values.length;
}
